woocommerce/woocommerce
# Fix a bug where woocommerce removes the current-menu-item class

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





`wc_nav_menu_item_classes` can currently remove the `current-menu-item` from a taxonomy menu item if its ID matches with the ID of the `page_for_posts`. This PR adds the menu_item type `page` as a condition.

Fixes #46060



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Note that we need to create a conflict between a category ID and a page ID to reproduce this issue. Since creating that conflict like that is cumbersome, we will directly edit the setting as part of testing instructions so that we can test.
1. Switch to a theme that allows modifying menu items directly, such as store front.
2. Add a new product category, or use an existing one. Note down it's ID, (to get the ID of category, open it for editing, and note the `tag_ID` in the URL).
3. Update the option for posts for page like so: `wp option set 'page_for_posts' XXX` where `XXX` is the category ID.
4. Add this category as a menu navigation item, by going to Appearance > Menu and add the Product Category as menu item. Note that if you don't see the Product Category as an option, you would have to enable by clicking on the Screen Options and enabling product categories.
<img width="1665" alt="Screenshot 2024-06-26 at 3 31 50 PM" src="https://github.com/woocommerce/woocommerce/assets/7571618/cd6258c6-79af-45a5-a557-86774f40470d">
5. Add this category in menu and save.
6. Go to frontend and click on this category from navigation bar.
7. Make sure that it displays as selected.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Fix a bug where woocommerce removes the current-menu-item class.

#### Comment 

</details>
